packages/beacon-transport-walletconnect/src/communication-client/WalletConnectCommunicationClient.ts

```diff
@@ -152,39 +152,64 @@ export class WalletConnectCommunicationClient extends CommunicationClient {
       throw new MissingRequiredScope(PermissionScopeMethods.GET_ACCOUNTS)
     }
 
-    console.log('#### Requesting public keys')
+    // TODO: Use public key from namespace
+    const accounts = this.getTezosNamespace(session.namespaces).accounts
+    const [_namespace, _chainId, _address] = accounts[0].split(':', 3)
 
-    if (message.network.type !== this.wcOptions.network) {
-      throw new Error('Network in permission request is not the same as preferred network!')
-    }
+    let publicKey: string | undefined
```

Comment:
This is where I should check if `_address` is not by any chance a public key, right?